[dagster-tableau] Add basic docs (#24609)
dagster-io/dagster: docs/content/integrations/tableau.mdx

---

> title: "Using Tableau"

Reviewer: ```suggestion
title: "Using Dagster with Tableau"
description: Represent your Tableau assets in Dagster
```

Update title based on Sandy's feedback in https://github.com/dagster-io/dagster/pull/24269

---

> # Using Tableau

Reviewer: ```suggestion
# Using Dagster with Tableau
```

<ExperimentalCallout />

This guide provides instructions for using Dagster with Tableau. Your Tableau assets, such as data sources, sheets, and dashboards, can be represented in the Dagster asset graph, allowing you to track lineage and dependencies between Tableau assets and upstream data assets you are already modeling in Dagster.

## What you'll learn

- How to represent Tableau assets in the Dagster asset graph.

- How to customize asset definition metadata for these Tableau assets.

<details>
  <summary>Prerequisites</summary>

- Familiarity with asset definitions and the Dagster asset graph

- Familiarity with Dagster resources - Familiarity with Tableau concepts, like data sources, sheets, and dashboards

- A Tableau site, either on Tableau Cloud or Tableau Server

- A connected app configured to access Tableau. For more information, see [Use Tableau Connected Apps for Application Integration](https://help.tableau.com/current/online/en-us/connected_apps.htm) in the Tableau documentation.

</details>

## Represent Tableau assets in the asset graph

> To load Tableau assets into the Dagster asset graph, you must first construct a Tableau resource, which allows Dagster to communicate with your Tableau workspace. The Tableau resource to create depends on your Tableau deployment type - `TableauCloudWorkspace` for Tableau Cloud and `TableauServerWorkspace` for Tableau Server. To connect to the Tableau workspace, you'll need to [configure a connected app with direct trust](https://help.tableau.com/current/online/en-gb/connected_apps_direct.htm) in Tableau, then supply your Tableau site information and connected app credentials to the resource. The Tableau resource uses the JSON Web Token (JWT) authentication to connect to the Tableau workspace.

Reviewer: ```suggestion
To load Tableau assets into the Dagster asset graph, you must first construct a Tableau resource, which allows Dagster to communicate with your Tableau workspace. The Tableau resource to create depends on your Tableau deployment type - use `TableauCloudWorkspace` if you are using Tableau Cloud or `TableauServerWorkspace` if you are using Tableau Server. To connect to the Tableau workspace, you'll need to [configure a connected app with direct trust](https://help.tableau.com/current/online/en-gb/connected_apps_direct.htm) in Tableau, then supply your Tableau site information and connected app credentials to the resource. The Tableau resource uses the JSON Web Token (JWT) authentication to connect to the Tableau workspace.
```

nit: suggestion to make it more clear that you use one or the other

Dagster can automatically load all data sources, sheets, and dashboards from your Tableau workspace. Call the `build_defs()` function, which returns a `Definitions` object containing all the asset definitions for these Tableau assets.

```python file=/integrations/tableau/representing-tableau-assets.py
from dagster_tableau import TableauCloudWorkspace, TableauServerWorkspace

from dagster import EnvVar

> # Connect to Tableau Cloud using the connected app credentials

Reviewer: Not blocking merging, but could be nice to toggle between the two using a tab group, e.g. https://github.com/dagster-io/dagster/blob/master/docs/content/concepts/assets/external-assets.mdx#L44
